### DISCUSSION TITLE
.env file

### DISCUSSION THREAD

**Discussion by iPagar** [railwayapp/nixpacks#321] [Q&A] [answered]
Hi!
Is there a way to load env vars from a file?

For example `nixpacks build ./path/to/app --envFile ./.env --name my-app`?

**Comment by coffee-cup** [comment, marked as answer]
There isn't at the moment, but would be a great feature to add! I think we could even do it automatically (if there is a `.env` file in the app root then load it)

**Reply by iPagar** [reply]
Yeah! Do it automatically would be great!

I found two ways of doing this now:
1. `nixpacks build . --name my-app --build-cmd 'set -a; source .env; set +a; npm run build'`
2. `nixpacks build . --name my-app --build-cmd 'export $(xargs < .env); npm run build'`

It's working perfectly! 😼

**Comment by Milo123459** [comment]
So like dotenv but without the package, good idea!

**Reply by coffee-cup** [reply]
I think we can handle it in a similar way to how the Procfile detection works. If there is a `.env` file found then parse and load all of the variables into the environment

**Reply by aarnphm** [reply]
We can provide both options, `--envFile` if users wants to provide different env file, and detect `.env` by default?
